### PR TITLE
vararg only the last free attribute

### PR DESCRIPTION
**Pull request by Graur**
Closed: #1511

---

## PR-Codex overview
This PR adds support for variadic arguments in EO, changes some test files and removes some unused files.

### Detailed summary
- Adds support for variadic arguments in EO
- Changes some test files
- Removes unused `middle-varargs.xsl`, `catches-middle-vararg.yaml` and `varargs-inside.yaml` files.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by Graur** [comment]
@volodya-lombrozo Please, take a look

File: eo-maven-plugin/src/test/java/org/eolang/maven/OptimizeMojoTest.java [hunk 1]
```diff
@@ -268,13 +268,13 @@ void stopsOnCritical(@TempDir final Path temp) throws Exception {
                     .result()
                     .get(
                         String.format(
-                            "target/%s/foo/x/main/02-duplicate-names.xml",
+                            "target/%s/foo/x/main/26-duplicate-names.xml",
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/test/java/org/eolang/maven/OptimizeMojoTest.java:271]: @Graur To be honest, I don't clearly understand why `26`. Does it some sort of the order?

File: eo-parser/src/main/antlr4/org/eolang/parser/Program.g4 [hunk 2]
```diff
@@ -61,7 +61,11 @@ abstraction
 attributes
   :
   LSQ
-  (attribute (SPACE attribute)*)?
+  (
+    (attribute (SPACE attribute)* (SPACE vararg)?)?
```
Comment (volodya-lombrozo) [inline eo-parser/src/main/antlr4/org/eolang/parser/Program.g4:65]: @Graur Looks like in some cases we can have a reverse varargs, like:
```eo
...b
```
https://github.com/objectionary/eo/blob/master/eo-parser/src/test/resources/org/eolang/parser/packs/varargs-both-ways.yaml

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

File: eo-parser/src/main/antlr4/org/eolang/parser/Program.g4 (hunk 2 above)
Comment (Graur) [reply eo-parser/src/main/antlr4/org/eolang/parser/Program.g4:65]: @volodya-lombrozo Yes, we can. But this lexema rule is only used for free attributes. And there is no rule for reverse vararg as a free attribute in EO. The reverse vararg is used to convert a vararg to an array.

File: eo-maven-plugin/src/test/java/org/eolang/maven/OptimizeMojoTest.java (hunk 1 above)
Comment (Graur) [reply eo-maven-plugin/src/test/java/org/eolang/maven/OptimizeMojoTest.java:271]: @volodya-lombrozo Yes, it's something like `xsl` file execution order

**Review by volodya-lombrozo — APPROVED** [review]
@Graur Looks good to me, thanks.
@yegor256 Could you take a look, please?

**Comment by Graur** [comment]
@yegor256 Please, have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**